mobile/src/stateMachine/views/healthData/healthScopes.tsx

```diff
@@ -73,6 +75,27 @@ export class HealthScopesView extends ViewState {
             },
         });
 
+    }
+
+    onAndroidSettings = () => {
+        AppController.Instance.PromptModal.openModal({
+            typeModal: 'positive',
+            title: 'Allow CompanionKit to access your Google Fit data',
+            message: 'Please grant access to Companion Kit.',
+            confirmText: 'Authorize Google Fit',
+            customView:
+                <View>
+                    <View style={styles.imageNextToText}>
+                        <Image source={Images.GoogleFitHealthPermPNG} style={{ width: 26, height: 26 }} />
+                        <Text style={[TextStyles.p1, styles.instructionText]}>{'Turn All Categories On'}</Text>
+                    </View>
+                </View>,
+            onConfirm: async () => {
+                AppController.Instance.PromptModal.closeModal();
+                await authAndroid();
```

Comment:
@MarkDavies0 there is a small fix needed here where the screen the user ends on after pressing confirm needs to refresh to show the updated auth. Or another option is to send the user to the homescreen after auth success

Reply:
@brendeny I think you need to `await authAndroid()` before calling `AppController.Instance.PromptModal.closeModal();` . Alternatively you could have the view react to the new auth status and re-render itself